Documentation/teaching/so2/grading.rst

```diff
@@ -0,0 +1,208 @@
+See also the next paragraph.
+
+It is not allowed to publish assignment solutions (even after the end of the course).
+If you find assignment solutions on GitHub or elsewhere, report them to the discussion list or privately to the laboratory assistant or course instructor.
+We reiterate that if you need clarification that you would address to older colleagues or other forums, StackExchange, or other sources, use the discussion list and the SO2 team.
+It is the safest and most honest way to solve problems.
+
+It is not allowed to transfer files between yourselves.
+In general, we recommend not to screen-share with another colleague, whether for inspiration or to help them with their assignment.
+Avoid testing an assignment on a colleague's system.
+There may be exceptions; you can help someone troubleshoot, but please ensure that it does not transition from "let's solve this problem together" to "let me solve your assignment for you".
+However, we recommend using the discussion list or the SO2 team to ask questions.
+
+5. Penalties for Copied Assignments
+-----------------------------------
+
+In general, we consider punitive measures as a last resort.
+As long as the assignment is completed individually, without problematic source code contribution from external sources, then it is not a copied assignment.
+
+The notion of a copied assignment refers to, without limitation, situations such as:
+
+    * Two assignments that are similar enough to draw this conclusion;
+    * Using source code from the internet that is an obvious solution to the assignment;
+    * Using pieces of code from another colleague;
+    * Accessing another colleague's code during the assignment;
+    * Modifying an existing assignment;
+    * Following another colleague's code;
+    * Direct assistance in completing the assignment (someone else wrote or dictated the code);
+    * Someone else wrote the assignment (voluntarily, for payment, or other benefits).
+    * If two assignments are considered copied, both the source and destination will be penalized equally, without discussions about who copied from whom and whose fault it is.
+
+.. warning::
+
+    Copying an assignment results in the elimination of points for the assignments completed up to that session.
+    Any assignment submitted until that session receives a score of 0 and cannot be resubmitted during the current academic year.
+    If there were instances of copied assignments during the semester, it will be possible to obtain points in the summer, for the September session, from assignments **not yet** submitted.
+    We reiterate that our goal is not and will not be penalization for copying.
+    We consider copying to be dishonest behavior that will be punished if it occurs.
+    However, our goal is to prevent copying; for this purpose, we offer support and resources from the team in all its forms (discussion list, face-to-face discussions with the SO2 team).
```

Comment:
```suggestion
    However, our goal is to prevent cheating; for this purpose, we offer support and resources from the team in all its forms (discussion list, face-to-face discussions with the SO2 team).
```